presto-main/src/main/java/com/facebook/presto/operator/scalar/MapConstructor.java

```diff
@@ -140,6 +150,10 @@ public static Block createMap(
                 mapBlockBuilder.closeEntry();
                 throw new PrestoException(INVALID_FUNCTION_ARGUMENT, "map key cannot be null");
             }
+            if (isIndeterminate(keyIndeterminate, mapType.getKeyType(), keyBlock, i)) {
```

Comment:
A PrestoException can be thrown when calling `indeterminate` MethodHandle (e.g. INVAILD_ARGUMENT`). 

We could inline `isIndeterminate` and handle this in one place:

```java
            try {
                if (!(boolean) keyIndeterminate.invoke(readNativeValue(mapType.getKeyType(), keyBlock, i), false)) {
                    throw new PrestoException(INVALID_FUNCTION_ARGUMENT, "map key cannot be indeterminate");
                }
            }
            catch (Throwable t) {
                // close block builder before throwing as we may be in a TRY() call
                // so that subsequent calls do not find it in an inconsistent state
                mapBlockBuilder.closeEntry();

                throwIfInstanceOf(t, PrestoException.class);
                throwIfUnchecked(t);
                throw new RuntimeException(t);
            }

```